### ISSUE TITLE
No values with option `pluck`

### ISSUE THREAD

**Issue by jorol** [gbv/PICA-Data#121]
When using option `pluck`, `pica_map` doesn´t return any matches, if the first value of the result set is `undefined`; 

```terminal
$ echo '[{"_id":"845005812","record":[["028C","","d","Katharina","a","Draheim","c","auf der","B","HerausgeberIn","4","edt"]]}]' | catmandu convert -v JSON to YAML --fix 'pica_map(028CXa,persons,split:1,nested_arrays:1,pluck:1);retain_field(persons);'
--- {}
...
converted 1 item
done
$ echo '[{"_id":"845005812","record":[["028C","","d","Katharina","a","Draheim","c","auf der","B","HerausgeberIn","4","edt"]]}]' | catmandu convert -v JSON to YAML --fix 'pica_map(028CaX,persons,split:1,nested_arrays:1,pluck:1);retain_field(persons);'
---
persons:
- - Draheim
  - ~
...
converted 1 item
done
```

https://github.com/gbv/PICA-Data/blob/main/lib/PICA/Path.pm#L135 should test, if array has any values:

```perl
next unless grep { defined $_ } @matched_subfields;
```

**Comment by jorol** [comment]
I've added a fix to branch "gh#121", but I'm not sure what the desired result of `pluck` in this case is. With my bugfix, an `undefined` value is added in the beginning of the subfield list for the non-existing subfield code :

```terminal
$ echo '[{"_id":"845005812","record":[["028C","","d","Katharina","a","Draheim","c","auf der","B","HerausgeberIn","4","edt"]]}]' | catmandu  -I /lib convert -v JSON to YAML --fix 'pica_map(028CXa,persons,split:1,nested_arrays:1,pluck:1);retain_field(persons);'
---
persons:
- - ~
  - Draheim
...
converted 1 item
done
``` 

As `Catmandu::Fix::marc_map` seems also be affected, we should discuss this with them.

**Comment by nichtich** [comment]
Fixed in dev branch.
